app/pages/SearchResultsPage/SearchResultsPage.tsx

```diff
@@ -34,22 +36,64 @@ interface SearchState extends ParsedQs {
 
 /** Wrapper component that handles state management, URL parsing, and external API requests. */
 export const SearchResultsPage = () => {
+  const [cookies] = useCookies(["googtrans"]);
+  const history = useHistory();
+  const { search } = useLocation();
   const { userLocation } = useAppContext();
   const [lastPush, setLastPush] = useState(Date.now());
-  const { search } = useLocation();
   const [expandList, setExpandList] = useState(false);
-
-  const searchState: SearchState = useMemo(
-    () => qs.parse(search.slice(1)),
-    [search]
-  );
+  const [searchState, setSearchState] = useState<SearchState>({});
   const [searchRadius, setSearchRadius] = useState(
     searchState?.configure?.aroundRadius ?? "all"
   );
+  // In cases where we translate a non-English query into English (to query Alglolia), we use this
+  // state var to cache the user's orginal, untranslated query to display on the results page
+  const [untranslatedQuery, setUntranslatedQuery] = useState("");
+  const [searchStateResolved, setSearchStateResolved] = useState(false);
+
+  useEffect(() => {
+    const queryParams = qs.parse(search.slice(1));
+    const { query } = queryParams;
+    setUntranslatedQuery(query as string);
+
+    // Google Translate determines translation source and target with a
+    // "googtrans" cookie. If the cookie exists, we assume that the
+    // the query should be translated into English prior to querying Algolia
+    const translationCookie = cookies.googtrans;
+
+    if (query && translationCookie) {
+      const [, targetLanguage] = translationCookie.split("/en/") as string;
+      if (targetLanguage !== "en") {
+        post("/api/translation/translate_text", {
+          text: query,
+          target_language: targetLanguage,
+        }).then((resp) =>
+          resp.json().then((body) => {
+            setSearchState({
+              ...queryParams,
+              query: body.result,
+            });
+            setSearchStateResolved(true);
+          })
+        );
+      }
+    } else {
+      setSearchState(queryParams);
+      setSearchStateResolved(true);
+    }
+  }, [cookies.googtrans, search, untranslatedQuery]);
+
+  if (!searchStateResolved) {
+    // In some case(s) -- e.g. when waiting to receive a translation from the API -- the search state
+    // is set asynchronously; thus, we want to wait until the search state resolves. Otherwise,
+    // our code will query Algolia twice - first with an empty string and secondly with the translated
```

Comment:
Not to nitpick grammar for grammar's sake, but did you mean "our code _would_ query Algolia twice"? As in, you're saying that if this code didn't have this early `return null`, then it would result in a UI flash and unnecessary call to Algolia?

Mostly asking because on my first read of these comments, I think I misread this as saying that we are intentionally acknowledging that we are going to cause a UI flash and unnecessary call to Algolia. Maybe you could reword this comment to be a bit more precise?